File: py/TotalCreditLines.py
"""
Created on May 30, 2013

@author:     Gregory Czajkowski

@copyright:  2013 Freedom. All rights reserved.

@license:    Licensed under the Apache License 2.0 http://www.apache.org/licenses/LICENSE-2.0

@contact:    gregczajkowski at yahoo.com
"""
import LoanEnum
import Filter


class TotalCreditLines(Filter.Filter):
    """
    class docs
    """
    sqlite_type = "INT"
    name = "total_acc"
    query = "(total_acc <= ?)"
    named_query = "(total_acc <= :total_acc)"

    def __init__(self, args, current=None):
        """
        Constructor
        """
        options = [10, 20, 30]

        Filter.Filter.__init__(self, args, options, current)

    def apply(self, loan, total_acc=LoanEnum.LOAN_ENUM_total_acc):
        return loan[total_acc] <= self.get_value()

File: cpp/LendingClub/TotalCreditLines.cpp
/*
Created on July 28, 2014

@author:     Gregory Czajkowski

@copyright:  2013 Freedom. All rights reserved.

@license:    Licensed under the Apache License 2.0 http://www.apache.org/licenses/LICENSE-2.0

@contact:    gregczajkowski at yahoo.com
*/

#include "TotalCreditLines.hpp"

using namespace lc;

const LCString TotalCreditLines::csv_name = "total_acc";
const LCString TotalCreditLines::name = "TotalCreditLines";
const FilterValueVector* TotalCreditLines::options = nullptr;
const Filter::Relation TotalCreditLines::relation = Filter::Relation::LESS_THAN_EQUAL;
